File: FlowerObjc/SourcesObjc/UIKit/SGLabel.m
//
//  SGLabel.m
//  SGLabel
//
//  Created by kingsic on 2020/12/19.
//  Copyright © 2020 kingsic. All rights reserved.
//

#import "SGLabel.h"

@implementation SGLabel

- (void)drawTextInRect:(CGRect)rect {
    CGRect tempRect = [self textRectForBounds:rect limitedToNumberOfLines:self.numberOfLines];
    [super drawTextInRect:tempRect];
}

- (CGRect)textRectForBounds:(CGRect)bounds limitedToNumberOfLines:(NSInteger)numberOfLines {
    CGRect textRect = [super textRectForBounds:bounds limitedToNumberOfLines:numberOfLines];
    textRect.origin.y = bounds.origin.y;
    return textRect;
}

@end

File: Flower/Source/UIKit/SGLabel.swift
//
//  SGLabel.swift
//  SGLabel
//
//  Created by kingsic on 2020/12/20.
//  Copyright © 2020 kingsic. All rights reserved.
//

import UIKit

class SGLabel: UILabel {

    override func drawText(in rect: CGRect) {
        let tempRect = self.textRect(forBounds: rect, limitedToNumberOfLines: self.numberOfLines)
        super.drawText(in: tempRect)
    }

    override func textRect(forBounds bounds: CGRect, limitedToNumberOfLines numberOfLines: Int) -> CGRect {
        var textRect = super.textRect(forBounds: bounds, limitedToNumberOfLines: numberOfLines)
        textRect.origin.y = bounds.origin.y
        return textRect
    }
    
    
}
